File: iosApp/iosApp/Features/Profile/components/AppSettings.swift
//
//  AppSettings.swift
//  iosApp
//
//  Created by velkonost on 20.09.2023.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import SharedSDK

struct AppSettings : View {
    
    let selectedTheme: UIMode
    let onThemeChanged: (UIMode) -> Void
    
    init(selectedTheme: UIMode, onThemeChanged: @escaping (UIMode) -> Void) {
        self.selectedTheme = selectedTheme
        self.onThemeChanged = onThemeChanged
    }
    
    var body: some View {
        SectionTitle(text: SharedR.strings().profile_app_settings_title.desc().localized())
        
        PrimaryBox {
            VStack {
                ThemingMenuItem(
                  selected: selectedTheme,
                  onClick: { value in
                      if value != selectedTheme {
                          onThemeChanged(value)
                      }
                  }
                )
                
//                MenuItem(
//                    title: SharedR.strings().profile_app_settings_notifications.desc().localized(),
//                    icon: SharedR.images().ic_notifications.toUIImage()!
//                ) {
//                    
//                }
            }
        }
    }
}

File: androidApp/features/profile/src/main/kotlin/com/velkonost/getbetter/android/features/profile/components/AppSettings.kt
package com.velkonost.getbetter.android.features.profile.components

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.velkonost.getbetter.android.features.profile.components.theming.ThemingMenuItem
import com.velkonost.getbetter.core.compose.components.PrimaryBox
import com.velkonost.getbetter.shared.core.model.profile.UIThemeMode
import com.velkonost.getbetter.shared.resources.SharedR
import dev.icerock.moko.resources.compose.stringResource

@Composable
fun AppSettings(
    modifier: Modifier = Modifier,
    selectedTheme: UIThemeMode,
    onThemeChanged: (UIThemeMode) -> Unit
) {
    SectionTitle(text = stringResource(resource = SharedR.strings.profile_app_settings_title))

    PrimaryBox {
//        Column {
        ThemingMenuItem(
            selected = selectedTheme,
            onClick = {
                if (it != selectedTheme) {
                    onThemeChanged.invoke(it)
                }
            }
        )
//            MenuItem(
//                modifier = modifier.padding(top = 4.dp),
//                title = stringResource(resource = SharedR.strings.profile_app_settings_notifications),
//                icon = painterResource(imageResource = SharedR.images.ic_notifications)
//            ) {
//
//            }
//        }
    }
}